src/Symfony/Component/Security/Core/Authentication/Provider/PreAuthenticatedAuthenticationProvider.php

```diff
@@ -54,7 +55,7 @@ public function authenticate(TokenInterface $token)
             throw new BadCredentialsException('No pre-authenticated principal found in request.');
         }
 
-        $user = $this->userProvider->loadUserByUsername($user);
+        $user = $this->userProvider->loadUserByUsername($user instanceof UserInterface ? $user->getUsername() : (string) $user);
```

Comment:
What about using `$token->getUsername()` instead? The code in the Abstract token already does exactly that check 😄  
```suggestion
        $user = $this->userProvider->loadUserByUsername($token->getUsername());
```